src/sickle.ts

```diff
@@ -222,7 +222,16 @@ class Annotator {
   private writeNode(node: ts.Node, skipComments: boolean = false) {
     if (node.getChildCount() == 0) {
       // Directly write complete tokens.
-      this.emit(skipComments ? node.getText() : node.getFullText());
+      if (skipComments) {
+        if (node.getFullStart() < node.getStart()) {
```

Comment:
Maybe extract into a helper function - `nodeHasAComment`, seems useful and not totally obvious from the node api